### DISCUSSION TITLE
sunbird.properties credentials required for sunbird_mobile_app

### DISCUSSION THREAD

**Discussion by israr-trigyn** [Sunbird-Ed/Community#402] [Sunbird ED configuration discussion] [answered]
Trying to run sunbird_mobile app from repository https://github.com/Sunbird-Ed/SunbirdEd-mobile-app.git

As per the readme file we required "sunbird.properties" credentials.

**Comment by Krishnaj20** [comment]
@swayangjit - Pls look into this query & respond.

@vinukumar-vs @chitranshu-keshav

**Reply by swayangjit** [reply]
@israr-trigyn There is a sample [sunbird.properties.sample](https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/blob/master/buildConfig/sunbird.properties.sample) which you have  update based upon your host and credentials.

**Reply by israr-trigyn** [reply]
Thank for responding,

Swayangjit i required 2 things basically 

1. google-services.json
2. sunbird.properties files

sunbird.properties file has lots of configurations variables, I have attached those variables on email can you please respond to that email along with the configuration and google-services.json file.

**Reply by swayangjit** [reply]
@israr-trigyn  This is for which environment?

**Reply by chitranshu-keshav** [reply]
@israr-trigyn can you please get back with the details

**Comment by ravishtrigyn** [comment]
@swayangjit This was a tenant's testing environment.

**Reply by swayangjit** [reply]
@ravishtrigyn  sunbird.properties varies from instance to instance. You can find out the documentation [here](https://ed.sunbird.org/development-resources/misc-pages/configurations-to-setup-mobile-app) .

**Comment by pushpamtrigyn** [comment]
Hi,

I am trying to run sunbird_mobile app from repository using branch release-5.1.0. 
**./build.sh** file runs successfully upto the following step:-

**NODE_OPTIONS=--max-old-space-size=4096 ionic cordova build android --prod --release --verbose --buildConfig ./buildConfig/build.json** 

**Steps**
- sunbird.properties file has been placed inside ./buildConfig folder
- build.json file has been placed inside ./buildConfig folder
- JKS file also placed inside ./buildConfig folder
- Installed native-run at one point "npm install native-run"

**Technical stack used**
- NodeJS 16.10.0
- Ionic 6.4.0
- Cordova 11.1.0
- Android SDK tools 30.0.3

**Results**
- APK builds successfully but shows white screen on actual mobile phone
- APK not installing on Android emulator
- APK shows blank page on web browser while running ionic serve, with "circular dependency warnings" on console

**A few logs which may be helpful**
<img width="718" alt="Logs 1" src="https://user-images.githubusercontent.com/127098111/224248892-b7110781-c940-4fff-b585-376f1765906a.png">
<img width="710" alt="Logs 2" src="https://user-images.githubusercontent.com/127098111/224248895-4e00965d-bc54-4773-85fd-82abe7661233.png">
<img width="705" alt="Logs 3" src="https://user-images.githubusercontent.com/127098111/224248897-9f810e44-4fb0-4fbf-8ba7-710cfcec425f.png">
<img width="691" alt="Logs 4" src="https://user-images.githubusercontent.com/127098111/224248898-3110771a-9e57-4f24-89b2-d73c098977b1.png">
<img width="693" alt="Logs 5" src="https://user-images.githubusercontent.com/127098111/224248900-7a149254-f538-4447-b273-e7966035b77d.png">
<img width="691" alt="Logs 6" src="https://user-images.githubusercontent.com/127098111/224248879-36fdb8c2-f859-478e-8a48-805aab98fa75.png">
<img width="635" alt="Logs 7" src="https://user-images.githubusercontent.com/127098111/224249814-5e81ff34-da61-4d48-a75d-47b92b017068.png">

@swayangjit @vinukumar-vs @rajeshkumaravel @anandvarada

**Reply by swayangjit** [reply]
@pushpamtrigyn  ionic-serve won't work here as the mobile app depends on the Cordova plugins. For the apk can you debug and share the log with us by inspecting it.

**Reply by israr-trigyn** [reply]
@swayangjit we tried running in real devices also but not success. we are getting below mentioned errors while running on real device

**ERROR LOG:-**

[cordova] sed: can't read platforms/android/cordova-plugin-badge/app-badge.gradle: No such file or directory
[cordova] Warning: unexpected element (uri:"", local:"base-extension"). Expected elements are <{}codename>,<{}layoutlib>,<{}api-level>
[cordova] 
[cordova] FAILURE: Build failed with an exception.
[cordova]
[cordova] * What went wrong:
[cordova] Execution failed for task ':app:processDevDebugGoogleServices'.
[cordova] > No matching client found for package name 'preprod.diksha.app'
[cordova]
[cordova] * Try:
[cordova] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[cordova]
[cordova] * Get more help at https://help.gradle.org
[cordova]
[cordova] BUILD FAILED in 4s
[cordova] Command failed with exit code 1: D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\gradlew cdvBuildDebug -b D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\build.gradle
[ERROR] An error occurred while running subprocess cordova.

 cordova.cmd build android --device exited with exit code 1.
        
  --------------------------------
   
**When we're trying to run "./build.sh" I noticed platforms not installing successfully, it throws below errors:**

Error: Command failed with exit code 1: D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\gradlew :app:bundleRelease -b D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\build.gradle
    at makeError (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\execa\lib\error.js:60:11)
    at handlePromise (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\execa\index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ProjectBuilder.build (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\cordova-android\lib\builders\ProjectBuilder.js:316:20)
[ERROR] An error occurred while running subprocess cordova.

I tried installing platform separately also but still getting same above error

**Reply by swayangjit** [reply]
@israr-trigyn  You have to update the google-service.json file as the error says  `There is no matching client for "preprod.diksha.app"`. The google-service.json file should have a client with id "preprod.diksha.app".

**Reply by israr-trigyn** [reply]
@swayangjit I did put a "client_id" under "project_info" object in google-services.json file and still the issue is same putting more logs below for your reference that im getting after running "./build.sh".

Exception in thread "main" java.io.FileNotFoundException: D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\gradle\wrapper\https\:\services.gradle.org\distributions\gradle-7.0.2-all.zip (The filename, directory name, or volume label syntax is incorrect)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
        at java.base/sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:86)       
        at java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:184)
        at org.gradle.wrapper.Download.downloadInternal(Download.java:109)
        at org.gradle.wrapper.Download.download(Download.java:89)
        at org.gradle.wrapper.Install$1.call(Install.java:83)
        at org.gradle.wrapper.Install$1.call(Install.java:63)
        at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:69)       
        at org.gradle.wrapper.Install.createDist(Install.java:63)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:109)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:66)
Command failed with exit code 1: D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\gradlew :app:bundleRelease -b D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\build.gradle
Error: Command failed with exit code 1: D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\gradlew :app:bundleRelease -b D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\build.gradle
    at makeError (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\execa\lib\error.js:60:11)
    at handlePromise (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\execa\index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ProjectBuilder.build (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\cordova-android\lib\builders\ProjectBuilder.js:316:20)
[ERROR] An error occurred while running subprocess cordova.

        cordova.cmd build android --release --verbose --buildConfig ./buildConfig/build.... exited with   
        exit code 1.

        Re-running this command with the --verbose flag may provide more information.
  ionic:utils-process onBeforeExit handler: 'process.exit' received +0ms
  ionic:utils-process onBeforeExit handler: running 2 functions +0ms
  ionic:utils-process error while killing process tree for 122856: Error: Command failed: taskkill /pid 122856 /T /F
  ionic:utils-process ERROR: The process "122856" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {    
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 122856 /T /F'
  ionic:utils-process } +332ms
  ionic:utils-process onBeforeExit handler: error from function: Error: Command failed: taskkill /pid 122856 /T /F
  ionic:utils-process ERROR: The process "122856" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {    
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 122856 /T /F'
  ionic:utils-process } +4ms
  ionic:utils-process error while killing process tree for 97540: Error: Command failed: taskkill /pid 97540 /T /F
  ionic:utils-process ERROR: The process "97540" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {    
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 97540 /T /F'
  ionic:utils-process } +3ms
  ionic:utils-process onBeforeExit handler: error from function: Error: Command failed: taskkill /pid 97540 /T /F
  ionic:utils-process ERROR: The process "97540" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {    
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 97540 /T /F'
  ionic:utils-process } +2ms
  ionic:utils-process processExit: exiting (exit code: 1) +4ms

> sunbirded-mobile-app@0.0.1 ionic-build
> ng build --eval--source-map --configuration=$npm_config_angular_configuration && cordova build android  

Option "evalSourceMap" is deprecated.
An unhandled exception occurred: Configuration '$npm_config_angular_configuration' is not set in the workspace.
See "C:\Users\ISRAR~1.AHM\AppData\Local\Temp\ng-pC0TBX\angular-errors.log" for further details.

TRIGYN+Israr.Ahmad@SPL2010 MINGW64 /d/Work Backups/Diksha_diksha.gov.in/SUNBIRD/SUNBIRD ED MOBILE/Mobile_app (release-5.1.0)
$

**Reply by swayangjit** [reply]
@israr-trigyn It seems "$" symbol is not working on windows machine. Can you update the script for build like following
For debug build update in package.json like following
`"ionic-build": "ng build --eval--source-map --configuration=debug && cordova build android"`
and execute `npm run ionic-build`

**Reply by israr-trigyn** [reply]
@swayangjit

**Still getting below errors while running "./build.sh":**

Error: Command failed with exit code 1: D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\gradlew :app:bundleRelease -b D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\platforms\android\build.gradle
    at makeError (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\execa\lib\error.js:60:11)
    at handlePromise (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\execa\index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ProjectBuilder.build (D:\Work Backups\Diksha_diksha.gov.in\SUNBIRD\SUNBIRD ED MOBILE\Mobile_app\node_modules\cordova-android\lib\builders\ProjectBuilder.js:316:20)
[ERROR] An error occurred while running subprocess cordova.

        cordova.cmd build android --release --verbose --buildConfig ./buildConfig/build.... exited with   
        exit code 1.

        Re-running this command with the --verbose flag may provide more information.
  ionic:utils-process onBeforeExit handler: 'process.exit' received +0ms
  ionic:utils-process onBeforeExit handler: running 2 functions +0ms
  ionic:utils-process error while killing process tree for 57152: Error: Command failed: taskkill /pid 57152 /T /F
  ionic:utils-process ERROR: The process "57152" not found.

  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Socket.<anonymous> (node:internal/child_process:450:11)
  ionic:utils-process     at Socket.emit (node:events:390:28)
  ionic:utils-process     at Pipe.<anonymous> (node:net:672:12) {
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 57152 /T /F'
  ionic:utils-process } +222ms
  ionic:utils-process onBeforeExit handler: error from function: Error: Command failed: taskkill /pid 57152 /T /F
  ionic:utils-process ERROR: The process "57152" not found.

  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Socket.<anonymous> (node:internal/child_process:450:11)
  ionic:utils-process     at Socket.emit (node:events:390:28)
  ionic:utils-process     at Pipe.<anonymous> (node:net:672:12) {
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 57152 /T /F'
  ionic:utils-process } +2ms
  ionic:utils-process error while killing process tree for 49484: Error: Command failed: taskkill /pid 49484 /T /F
  ionic:utils-process ERROR: The process "49484" not found.

  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {    
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 49484 /T /F'
  ionic:utils-process } +4ms
  ionic:utils-process onBeforeExit handler: error from function: Error: Command failed: taskkill /pid 49484 /T /F
  ionic:utils-process ERROR: The process "49484" not found.

  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:397:12)
  ionic:utils-process     at ChildProcess.emit (node:events:390:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1064:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {    
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 49484 /T /F'
  ionic:utils-process } +1ms
  ionic:utils-process processExit: exiting (exit code: 1) +1ms

> sunbirded-mobile-app@0.0.1 ionic-build
> ng build --eval--source-map --configuration=debug && cordova build android


**tried running "npm run ionic-build"** also but in this case same above error occuring  "There is no matching client for "preprod.diksha.app"" but this time it showing "in.gov.diksha.app" not "preprod.diksha.app"

**Reply by swayangjit** [reply]
@israr-trigyn  You have to add "in.gov.diksha.app"  client in your google-service.json.

**Reply by israr-trigyn** [reply]
in above reply you asked me to put "preprod.diksha.app" so I did put "preprod.diksha.app" value in "client_id" variable under "project_info" object in google-services.json

**Replay from @swayangjit :-** "You have to update the google-service.json file as the error says There is no matching client for "preprod.diksha.app". The google-service.json file should have a client with id "preprod.diksha.app"."

I followed the steps you mentioned in above reply. and i did try changing the "preprod.diksha.app" value with "in.gov.diksha.app" in the same "client_id" variable of "project_info" object. but no success

can you please guide where i have to put "in.gov.diksha.app"? coz in "project_info" object i already created a "client_id" with value  "preprod.diksha.app", check the attached screen below

<img width="957" alt="screen-discussion" src="https://user-images.githubusercontent.com/125034682/225858292-e6c1a3ac-47d9-4846-9d40-8bbcf7f85a37.png">

**Reply by Krishnaj20** [reply]
@anandvarada - this discussion thread is ongoing for a very long time. Can you facilitate a sync call between the requestor & @swayangjit to close this? Thanks

@madhucr

**Reply by swayangjit** [reply]
@israr-trigyn  How many instances that you want to build those many numbers of client you have to put it in the google service.json file . I will create a sample google-service.json file and share in few moment.

**Reply by swayangjit** [reply]
@israr-trigyn Suppose you have 3 variants or instances/productFlavour of your app mentioned in the [build-extras.gradle](https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/blob/24d87f0564bb68c07ec9a919decb7783316a6010/buildConfig/build-extras.gradle#L102) then your google-services.json should have 3 clients like following
`"client": [
  {
    "client_info": {
      "android_client_info": {
        "package_name": "org.sunbird.app.dev"
      }
    }
    .
    .
    .
    ],
    "services": {}
  },
  {
    "client_info": {
      "android_client_info": {
        "package_name": "org.sunbird.app.prod"
      }
    },
    .
    .
    .
    "services": {}
  },
  {
    "client_info": {
      "android_client_info": {
        "package_name": "org.sunbird.app.staging"
      }
    },
    .
    .
    .
    "services": {}
]
`

**Reply by israr-trigyn** [reply]
@swayangjit 

is this how client info should look for each instances?

"client_info": {
        "mobilesdk_app_id": "1:123456789000:android:f1bf012572b04063",
        "client_id": "in.gov.diksha.app",
        "client_type": 1,
        "android_client_info": {
          "package_name": "in.gov.diksha.app", 
          "certificate_hash": []
        }
      },

**Reply by swayangjit** [reply]
@israr-trigyn  I assume you have both pre-prod and prod instances. Ideally, it should have both clients preprod and  prod.

**Reply by israr-trigyn** [reply]
@swayangjit 

yes added both instances in the google service file:

"client_info": {
        "mobilesdk_app_id": "1:123456789000:android:f1bf012572b04063",
        "client_id": "preprod.diksha.app",
        "client_type": 1,
        "android_client_info": {
          "package_name": "preprod.diksha.app",
          "certificate_hash": []
        }
      },
      "client_info": {
        "mobilesdk_app_id": "1:123456789000:android:f1bf012572b04063",
        "client_id": "in.gov.diksha.app",
        "client_type": 2,
        "android_client_info": {
          "package_name": "in.gov.diksha.app",
          "certificate_hash": []
        }
      },
      
      **still getting below errors**
      
      
<img width="760" alt="image" src="https://user-images.githubusercontent.com/125034682/226317875-7d697ec0-8f1e-4639-99b1-1f6fd7b000d7.png">

**Reply by swayangjit** [reply]
@israr-trigyn  You might have added "in.gov.diksha.app" after your build.sh. Now you can do 2 things
1. Copy the google-service.json file available in root folder and replace the same in `SunbirdEd-mobile-app/platforms/android/app` folder(Preferred solution)
2. Remove the sb-cordova-plugin-fcm plugin and add it again
`ionic cordova plugin rm sb-cordova-plugin-fcm`
`ionic cordova plugin add https://github.com/Sunbird-Ed/sb-cordova-plugin-fcm.git#release-5.0.2`

**Reply by israr-trigyn** [reply]
Hi @swayangjit,

First method didn't work
but in second methods previous error gone but getting another error which is not understandable, see below:

<img width="728" alt="image" src="https://user-images.githubusercontent.com/125034682/226515912-70cabe02-0ca2-4a2f-b18a-4ceeb57e756c.png">

Are you familiar with this error??

**Reply by swayangjit** [reply]
@israr-trigyn Lets catchup on the issues that you are facing...

**Reply by israr-trigyn** [reply]
that will be really helpful, you can connect with me on israr.ahmad@trigyn.com

**Reply by Krishnaj20** [reply]
@chitranshu-keshav - Pls facilitate the call, in case if it hasn't happened till now.

**Reply by israr-trigyn** [reply]
@Krishnaj20 the issue is resolved, thanks to @swayangjit that he connected and helped me in resolving the issue, and i was still wondering if there is any alternative to run this app in "--livereload" mode

**Reply by swayangjit** [reply]
@israr-trigyn As of now the current live reload is not working as expected. So any changes to your code you have to run `npm run ionic-build`

**Comment by swayangjit** [comment, marked as answer]
@israr-trigyn To summarize the issue, After placing the proper google-service.json file and sunbird.properties file, the build.sh started working and generated the apk , regarding the live-reload mode right now it's not working but there are alternatives suggested and if you have any solution then feel free to contribute it back.

I hope the solutions provided resolved all your queries.
